site/app/controllers/HomePageController.php

```diff
@@ -133,6 +134,76 @@ function(Course $course) {
         );
     }
 
+    /**
+     * @Route("/home/new_course", methods={"POST"})
+     * @Route("/api/courses", methods={"POST"})
+     */
+    public function createCourse() {
+        $user = $this->core->getUser();
+        if (is_null($user) || $user->getAccessLevel() !== User::LEVEL_FACULTY) {
+            return new Response(
+                JsonResponse::getFailResponse("You don't have access to this endpoint."),
+                new WebResponse("Error", "errorPage", "You don't have access to this page.")
+            );
+        }
+
+        $semester = $_POST['semester'];
+        $course_title = $_POST['course_title'];
+        $head_instructor = $_POST['head_instructor'];
```

Comment:
Is the 'head instructor' copied from the template course (like how we copy the group from the template course)?  Or is the head instructor set to be the person who created the course?  

It should be the person who created the course.  A course might be 'team taught' in the fall with instructors A & B, where A was arbitrarily chosen as the owner for the files.  Then if instructor B creates the course in the spring and teaches the course alone, they should own the files.

Reply:
The head instructor is set to be the person who created the course.

There's a problem though... The `head_instructor` shouldn't be `POST`ed because it can be randomly modified by the `POST`er who may be one of faculty. Will change.

Reply:
Do we need to rename the `base_course` to `template_course`? I am not sure about the naming.

Reply:
naming isn't significant right now...  it's fine as it is and we can tweak it later if we change our mind.